<launch>
  <arg name="gui" default="true"/>

  <include file="$(find jsk_pcl_ros)/sample/sample_grid_sampler.launch">
    <arg name="gui" value="false"/>
  </include>

  <node name="mask_image_generator"
        pkg="nodelet" type="nodelet"
        args="load jsk_perception/MaskImageGenerator sample_manager">
    <remap from="~input" to="/kinect_head/rgb/image_rect_color"/>
    <rosparam>
      offset_x: 180
      offset_y: 160
      width: 20
      height: 50
    </rosparam>
  </node>

  <node name="mask_image_cluster_filter"
        pkg="nodelet" type="nodelet"
        args="load jsk_pcl/MaskImageClusterFilter sample_manager">
    <remap from="~input" to="/kinect_head/depth_registered/points"/>
    <remap from="~target" to="grid_sampler/output"/>
    <remap from="~input/camera_info" to="/kinect_head/rgb/camera_info"/>
    <remap from="~input/mask" to="mask_image_generator/output"/>
  </node>

  <group if="$(arg gui)">
    <node name="cluster_point_indices_decomposer"
          pkg="nodelet" type="nodelet"
          args="load jsk_pcl/ClusterPointIndicesDecomposer sample_manager">
      <remap from="~input" to="/kinect_head/depth_registered/points"/>
      <remap from="~target" to="grid_sampler/output"/>
    </node>

    <node name="extract_indices"
          pkg="nodelet" type="nodelet"
          args="load jsk_pcl/ExtractIndices sample_manager">
      <remap from="~input" to="/kinect_head/depth_registered/points"/>
      <remap from="~indices" to="mask_image_cluster_filter/output"/>
      <rosparam>
        keep_organized: true
      </rosparam>
    </node>

    <node name="rviz"
          pkg="rviz" type="rviz"
          args="-d $(find jsk_pcl_ros)/sample/rviz/mask_image_cluster_filter.rviz"/>
  </group>
</launch>
